Adding note about `node-id` being generated when unset (#24101)
hashicorp/vault: website/content/docs/configuration/storage/raft.mdx

  This value can be overridden by setting the `VAULT_RAFT_PATH` environment variable.

- `node_id` `(string: "")` - The identifier for the node in the Raft cluster.

>   This value can be overridden by setting the `VAULT_RAFT_NODE_ID` environment
>   variable.  If left unset, a random GUID will be generated by the Vault
>   instance during initialization and written to `data/node-id` in the directory
>   specified in `path`.

Reviewer: ```suggestion
  You can override `node_id` with the `VAULT_RAFT_NODE_ID` environment
  variable.  When `VAULT_RAFT_NODE_ID` is unset, Vault assigns a random
  GUID during initialization and writes the value to `data/node-id` in the
  directory specified by the `path` parameter.
```

Style correction: write in active voice, avoid "this" as a pronoun
